### PR TITLE
make mailtexts robust and testable

### PR DESCRIPTION
**Pull request by pascalfree**
Using `generate_testdata` to ensure that data needed for mailtexts exists.
The transition is then rolled back, to leave the database as it was.

# Release Notes
## Admin Features
* The mailtexts command now works regardless of the available objects in the database.

### CODE REVIEW

**Review by motu55 — COMMENTED** [review]
I think this is dangerous because if someone messes this code with the transaction you fill the productive databse with test stuff
a productive database should never be filled with "nonsense"
a better way is to create the objects without saving them

**Comment by pascalfree** [comment]
Any code is dangerous when messed up. And this would not even cause data loss and the "mess" would be very limited and predictable"
It just feels unsafe, because it's touching the database. ;-)

Saving is required to make the foreign keys work.

File: juntagrico/management/commands/mailtexts.py [hunk 1]
```diff
@@ -14,13 +16,16 @@
...
+        with transaction.atomic(durable=True):
+            # generate temporary test data to ensure that required objects are available
+            management.call_command('generate_testdata')
+            subscription = Subscription.objects.all()[0]
+            shares = Share.objects.all()[:2]
+            job = RecuringJob.objects.all()[0]
+            member, co_member = Member.objects.filter(MemberDao.has_future_subscription())[:2]
+            member_wo_subs = Member.objects.filter(subscriptionmembership__isnull=True)[0]
+            depot = Depot.objects.all()[0]
+            transaction.set_rollback(True)  # force rollback
```
Comment (motu55) [inline juntagrico/management/commands/mailtexts.py:28]: assume db crashes before the rollback is forced. the testdata might be found in a redo log of the db and be aplied
better way is to check if a correponding object exists in the db and if not use the new constructor without saving to create one

**Review by motu55 — CHANGES_REQUESTED** [review]

File: juntagrico/management/commands/mailtexts.py (hunk 1 above)
Comment (pascalfree) [reply juntagrico/management/commands/mailtexts.py:28]: foreign keys don't work without saving, at least not many to many and at least not the last time I tried that. So some of the templates fail.
Is this really how databases deal with uncompleted transactions after a crash? I would doubt it. Couldn't find any resource that supports yours nor my theory.
In any case I might try generating a temporary (in-memory) database to fill it with usable data for the email texts and use that. https://forum.djangoproject.com/t/using-an-in-memory-sqlite-database/472 or directly use the django test framework to achieve the same.

**Pull request merged**